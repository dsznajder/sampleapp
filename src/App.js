import React, { Component } from 'react'
import { connect } from 'react-redux'

import logo from './logo.svg'
import './App.css'
import { addTodo, removeTodo } from './store'
import TodoList from './components/TodoList'
import AddTodo from './components/AddTodo'

class App extends Component {
  addTodo = text => this.props.addTodo(text)
  removeTodo = todo => this.props.removeTodo(todo)

  render() {
    const { todos } = this.props

    return (
      <div className="App">
        <div className="App-header">
          <img alt="logo" className="App-logo" src={logo} />
          <h2>Welcome to React Example</h2>
        </div>
        <div className="App-intro">
          <AddTodo addTodo={this.addTodo} />
          <TodoList removeTodo={this.removeTodo} todos={todos} />
          <div>Count: {todos.length}</div>
        </div>
      </div>
    )
  }
}

const mapStateToProps = ({ todos }) => ({ todos })

const mapDispatchToProps = {
  addTodo,
  removeTodo,
}

export default connect(mapStateToProps, mapDispatchToProps)(App)
