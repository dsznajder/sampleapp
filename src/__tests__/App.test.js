import React from 'react'
import renderer from 'react-test-renderer'

import { App, mapStateToProps } from '../App'

describe('<App />', () => {
  const defaultProps = {
    addTodo: jest.fn(),
    removeTodo: jest.fn(),
    todos: [],
  }

  const wrapper = renderer.create(<App {...defaultProps} />)

  test('render', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
