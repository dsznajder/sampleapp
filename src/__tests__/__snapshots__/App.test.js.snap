// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> render 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h2>
      Welcome to React Example
    </h2>
  </div>
  <div
    className="App-intro"
  >
    <div>
      <input
        onChange={[Function]}
        onKeyDown={[Function]}
        value={undefined}
      />
      <button
        disabled={true}
        onClick={[Function]}
      >
        Add
      </button>
    </div>
    <ul />
    <div>
      Count: 
      0
    </div>
  </div>
</div>
`;
